pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +34,33 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+const inClusterNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+var inClusterNamespace string
+
+// GetInClusterNamespace returns the namespace by which the controller is running under.
+func GetInClusterNamespace() (string, error) {
```

Comment:
Can we get some tests in place?

Reply:
Done, though it's a bit clunky because of having of envtest not running the controller in a real cluster.